integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java

```diff
@@ -0,0 +1,272 @@
+      indexerSpec = indexerSpec.replaceAll("%%ZOOKEEPER_SERVER%%", config.getZookeeperHosts());
+      indexerSpec = indexerSpec.replaceAll("%%GROUP_ID%%", Long.toString(System.currentTimeMillis()));
+      indexerSpec = indexerSpec.replaceAll(
+          "%%SHUTOFFTIME%%",
+          new DateTime(
+              System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(
+                  2
+                  * MINUTES_TO_SEND
+              )
+          ).toString()
+      );
+      LOG.info("indexerFile: [%s]\n", indexerSpec);
+    }
+    catch (Exception e) {
+      // log here so the message will appear in the console output
+      LOG.error("could not read indexer file [%s]", INDEXER_FILE);
+      throw new ISE(e, "could not read indexer file [%s]", INDEXER_FILE);
+    }
+
+    // start indexing task
+    taskID = indexer.submitTask(indexerSpec);
+    LOG.info("-------------SUBMITTED TASK");
+
+    // set up kafka producer
+    Properties properties = new Properties();
+    properties.put("metadata.broker.list", config.getKafkaHost());
+    LOG.info("kafka host: [%s]", config.getKafkaHost());
+    properties.put("serializer.class", "kafka.serializer.StringEncoder");
+    properties.put("request.required.acks", "1");
+    properties.put("producer.type", "async");
+    ProducerConfig producerConfig = new ProducerConfig(properties);
+    Producer<String, String> producer = new Producer<String, String>(producerConfig);
+
+    DateTimeZone zone = DateTimeZone.forID("UTC");
+    // format for putting into events
+    DateTimeFormatter event_fmt = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+
+    DateTime dt = new DateTime(zone); // timestamp to put on events
+    dtFirst = dt;            // timestamp of 1st event
+    dtLast = dt;             // timestamp of last event
+    // stop sending events when time passes this
+    DateTime dtStop = dtFirst.plusMinutes(MINUTES_TO_SEND).plusSeconds(30);
+
+    // these are used to compute the expected aggregations
+    int added = 0;
+    int num_events = 0;
+
+    // send data to kafka
+    while (dt.compareTo(dtStop) < 0) {  // as long as we're within the time span
+      LOG.info("sending event at [%s]", event_fmt.print(dt));
+      num_events++;
+      added += num_events;
+      // construct the event to send
+      String event = String.format(
+          event_template,
+          event_fmt.print(dt), num_events, 0, num_events
+      );
+      LOG.debug("event: [%s]", event);
+      try {
+        // Send event to kafka
+        KeyedMessage<String, String> message = new KeyedMessage<String, String>(TOPIC_NAME, event);
+        producer.send(message);
+      }
+      catch (Exception ioe) {
+        Throwables.propagate(ioe);
+      }
+
+      try {
+        Thread.sleep(DELAY_BETWEEN_EVENTS_SECS * 1000);
+      }
+      catch (InterruptedException ex) { /* nothing */ }
+      dtLast = dt;
+      dt = new DateTime(zone);
+    }
+
+    producer.close();
+
+    // wait for segments
```

Comment:
we should also probably query the realtime task before segments are handed off in order to make sure realtime task are able to serve queries properly.